File: src/main/java/com/yerti/core/utils/LocationUtils.java
package com.yerti.core.utils;


import com.yerti.core.config.Config;
import org.bukkit.Bukkit;
import org.bukkit.Location;

public class LocationUtils {

    public static String serializeLocation(Location loc) {
        return loc.getWorld() + ";;" + loc.getX() + ";;" + loc.getY() + ";;" + loc.getZ();
    }

    public static void saveLocation(Config config, Location loc, String path) {
        config.set(path + ".world", loc.getWorld().getName());
        config.set(path + ".x", loc.getX());
        config.set(path + ".y", loc.getY());
        config.set(path + ".z", loc.getZ());
        config.save();
    }

    public static Location getLocation(Config config, String path) {
        if (config.getConfigurationSection(path) == null) return null;

        String world = config.getTranslation(path + ".world");
        double x = config.getDouble(path + ".x");
        double y = config.getDouble(path + ".y");
        double z = config.getDouble(path + ".z");
        float yaw = (float) config.getDouble(path + ".yaw");
        float pitch = (float) config.getDouble(path + ".pitch");
        Location newLocation = new Location(Bukkit.getWorld(world), x, y, z);
        newLocation.setYaw(yaw);
        newLocation.setPitch(pitch);
        return newLocation;
    }

    public static Location deserializeLocation(String string) {
        String[] split = string.split(";;");

        return new Location(Bukkit.getWorld(split[0]), Double.parseDouble(split[1]), Double.parseDouble(split[2]), Double.parseDouble(split[3]));
    }


}

File: src/main/java/com/yerti/core/utils/LocationUtils.kt
package com.yerti.core.utils

import com.yerti.core.config.Config
import org.bukkit.Bukkit
import org.bukkit.Location

object LocationUtils {
    @JvmStatic
    fun serializeLocation(loc: Location): String {
        return loc.world.toString() + ";;" + loc.x + ";;" + loc.y + ";;" + loc.z
    }

    fun saveLocation(config: Config, loc: Location, path: String) {
        config["$path.world"] = loc.world.name
        config["$path.x"] = loc.x
        config["$path.y"] = loc.y
        config["$path.z"] = loc.z
        config.save()
    }

    fun getLocation(config: Config, path: String): Location? {
        if (config.getConfigurationSection(path) == null) return null
        val world = config.getTranslation("$path.world")
        val x = config.getDouble("$path.x")
        val y = config.getDouble("$path.y")
        val z = config.getDouble("$path.z")
        val yaw = config.getDouble("$path.yaw").toFloat()
        val pitch = config.getDouble("$path.pitch").toFloat()
        val newLocation = Location(Bukkit.getWorld(world), x, y, z)
        newLocation.yaw = yaw
        newLocation.pitch = pitch
        return newLocation
    }

    fun deserializeLocation(string: String): Location {
        val split = string.split(";;").toTypedArray()
        return Location(Bukkit.getWorld(split[0]), split[1].toDouble(), split[2].toDouble(), split[3].toDouble())
    }
}